integrations/spark/openhouse-spark-runtime/build.gradle

```diff
@@ -61,23 +53,57 @@ dependencies {
     transitive = false
   }
   implementation("org.apache.iceberg:iceberg-spark-runtime-3.1_2.12:" + icebergVersion)
+}
 
-  generateGrammarSource {
-    maxHeapSize = "64m"
-    arguments += ['-visitor', '-package', 'com.linkedin.openhouse.spark.sql.catalyst.parser.extensions']
+// Define antlr directories for source generation
+ext {
+  antlrPackageDirPrefix="com/linkedin/openhouse/spark/sql/catalyst/parser/extensions/"
+  antlrMainDir="${projectDir}/src/main/antlr/${antlrPackageDirPrefix}"
+  antlrMainGeneratedSrcDir="${project.buildDir}/generated-src/antlr/main/"
+}
+
+// Set source for antlr generated directory
+sourceSets {
+  main {
+    java {
+      srcDirs antlrMainGeneratedSrcDir
+    }
   }
 }
 
+// Task to generate java sources using Antlr tool
+task runAntlr(type:JavaExec) {
+  inputs.dir antlrMainDir
+  outputs.dir antlrMainGeneratedSrcDir
+
+  main = "org.antlr.v4.Tool"
+  args = ["${antlrMainDir}/OpenhouseSqlExtensions.g4", 
+          "-visitor", 
+          "-o", "${antlrMainGeneratedSrcDir}/${antlrPackageDirPrefix}", 
+          "-package", "com.linkedin.openhouse.spark.sql.catalyst.parser.extensions"]
+  maxHeapSize = "64m"
+  classpath = configurations.antlr
+}
+
+compileJava.dependsOn runAntlr
 
 shadowJar {
   dependencies {
     exclude("javax/**")
   }
-
+  
   configurations = [project.configurations.fatJarPackagedDependencies]
   mergeServiceFiles()
   archiveClassifier.set('uber')
   zip64 true
 }
 
-jar.enabled=true
+jar {
+  enabled=true
+  duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+}
+
+sourcesJar {
+  duplicatesStrategy = DuplicatesStrategy.EXCLUDE
```

Comment:
why was this needed?

Reply:
Latest changes shows that this strategy is not needed. Hence, removed.